cmd/contour/serve.go

```diff
@@ -396,13 +396,13 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 			ResponseHeadersPolicy: &responseHeadersPolicy,
 		},
 	}
-	if clients.ResourcesExist(k8s.GatewayAPIResources()...) {
+	if ctx.Config.GatewayConfig != nil {
```

Comment:
What do we want to do if a gateway is configured, but the gateway APIs aren't installed in the cluster? Right now I believe we'll log some warnings about not finding the resources but allow Contour to start up (ignoring the gateway config). Might be a better UX to exit, to make it obvious that something is wrong.

Reply:
I redid that logic to incorporate your suggestion. 

- If the APIs are not installed & gateway is not configured, all is well
- If the APIs are not installed & gateway IS configured, fatal
- If the APIs are installed & gateway is not configured, all is well